frontend/src/context/ThemeContext.tsx

```diff
@@ -0,0 +1,38 @@
+import React, {createContext, useContext, useState, useEffect} from 'react';
+
+interface ThemeContextType {
+  isDarkMode: boolean;
+  toggleTheme: () => void;
+}
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
+
+export const useTheme = () => {
+  const context = useContext(ThemeContext);
+  if (context === undefined) {
+    throw new Error('useTheme must be used within a ThemeProvider');
+  }
+  return context;
+};
+
+export const ThemeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const [isDarkMode, setIsDarkMode] = useState(() => {
+    return localStorage.getItem('isDarkMode') === 'true';
```

Comment:
small nit: Preferable to namespace the key in localStorage to prefix with ziya, something like "ZIYA_IS_DARK_MODE"